noxfile.py

```diff
@@ -130,24 +131,40 @@ def default(session, install_grpc=True, prerelease=False, install_auth_aio=False
     )
 
     constraints_dir = str(CURRENT_DIRECTORY / "testing")
-
+    constraints_type = "async-rest-" if install_async_rest else ""
     if prerelease:
         install_prerelease_dependencies(
-            session, f"{constraints_dir}/constraints-{PYTHON_VERSIONS[0]}.txt"
+            session,
+            f"{constraints_dir}/constraints-{constraints_type}{PYTHON_VERSIONS[0]}.txt",
         )
         # This *must* be the last install command to get the package from source.
-        session.install("-e", ".", "--no-deps")
-    else:
         session.install(
-            "-e",
-            ".[grpc]" if install_grpc else ".",
-            "-c",
-            f"{constraints_dir}/constraints-{session.python}.txt",
+            "-e", "." + ("[async_rest]" if install_async_rest else ""), "--no-deps"
         )
+    else:
+        constraints_file = (
+            f"{constraints_dir}/constraints-{constraints_type}{session.python}.txt"
+        )
+        # fall back to standard constraints file
+        if not pathlib.Path(constraints_file).exists():
```

Comment:
@vchudnov-g shall we get rid of this section? Given that we've added a new constraints file, I think this is extra work and probably not needed. WDYT?

Reply:
As discussed offline, the constraints file will only enforce the versions but not install the requirements, so this is needed.